django_guid/middleware.py

```diff
@@ -22,95 +25,96 @@ def __init__(self, get_response):
         response = self.get_response(request)
-
         # Delete the current request to avoid memory leak
         self.__class__.del_guid()
-
         return response
 
     @classmethod
     def get_guid(cls, default=None):
         """
-        Fetches the GUID to the current thread, or the optionally
-        provided default if there is no current GUID.
-        :return: Fetches the GUID for the thread
+        Fetches the GUID of the current thread, from _guid.
+        If no value has been set for the current thread yet, we return a default value.
+        :default: Optional value to return if no GUID has been set on the current thread.
+        :return: GUID or default.
         """
         return cls._guid.get(threading.current_thread(), default)
 
     @classmethod
-    def set_guid(cls, guid):
+    def set_guid(cls, guid: str) -> None:
         """
-        Sets the GUID to the thread
-        :param guid: A guid/uuid4
+        Assigns a GUID to the thread.
+        :param guid: str
+        :return: None
         """
         cls._guid[threading.current_thread()] = guid
 
     @classmethod
-    def del_guid(cls):
+    def del_guid(cls) -> None:
         """
         Delete the guid that was stored for the current thread.
+        :return: None
         """
         cls._guid.pop(threading.current_thread(), None)
 
     @staticmethod
     def _generate_guid() -> str:
         """
-        Generates an UUIDv4/GUID as a string
+        Generates an UUIDv4/GUID as a string.
         :return: GUID
         """
         return uuid.uuid4().hex
 
     @staticmethod
     def _validate_guid(original_guid: str) -> bool:
         """
-        Validates a GUID
-        :param original_guid: GUID to check
+        Validates a GUID.
+        :param original_guid: string to validate
         :return: bool
         """
         try:
-            guid_value = uuid.UUID(original_guid, version=4)
+            return original_guid == uuid.UUID(original_guid, version=4).hex
         except ValueError:
             return False
-        return original_guid == guid_value.hex
 
     def _get_correlation_id_from_header(self, request: HttpRequest) -> str:
         """
         Returns either the provided GUID or a new one,
-        depending on if it's a valid GUID or not and the specified settings
+        depending on if the provided GUID is valid, and the specified settings
         :param request: HttpRequest object
         :return: GUID
         """
-        given_guid = str(request.headers.get('Correlation-ID'))
-        if settings.VALIDATE_GUID is True and self._validate_guid(given_guid):
+        given_guid = str(request.headers.get(settings.GUID_HEADER_NAME))
+        if not settings.VALIDATE_GUID:
             return given_guid
-        elif settings.VALIDATE_GUID is False:
+        elif settings.VALIDATE_GUID and self._validate_guid(given_guid):
             return given_guid
-        return self._generate_guid()
+        else:
```

Comment:
Could remove the `else`. :)

Reply:
Yes, it can definitely be removed; I only added it because it took me a little while to understand the flow. I thought adding it might help highlight that this is the third and last logical option to be handled, rather than code to be defaulted to when none of the previous checks are triggered. In a closed control flow, perhaps an else statement is syntactically clearer?

Reply:
Sure, the clearer the code the better.